### PR TITLE
Add config to set OpenShift build log level

### LINKED ISSUE #30277: I can't set "log category" for Quarkus-openshift deployment classes
### Describe the bug

Quarkus 2.15.0.Final

I am trying to deploy an application into Openshift and I would like to see all the DEBUG log traces of  "io.quarkus.container.image.openshift.deployment" , but seems that this config only applies to the application itself not to the deployment process. 

I think that would be great to be able to debug the deployment extensions. 

FYI: @llowinge 

### Expected behavior

some "io.quarkus.container.image.openshift.deployment" DEBUG log traces.

### Actual behavior

No logs debug traces

### How to Reproduce?

application.properties
```
quarkus.log.category."io.quarkus.container.image.openshift.deployment".level=DEBUG
quarkus.log.category."io.quarkus.container.image.openshift.deployment".min-level=DEBUG
```
```
quarkus create app org.acme:openshift-quickstart \
    --extension='resteasy-reactive,openshift'
```

```
cd openshift-quickstart
```

```
mvn clean package -DskipTests -Dquarkus.kubernetes.deploy=true -Dquarkus.kubernetes-client.trust-certs=true
```

### Output of `uname -a` or `ver`

Linux fedora 6.0.16-300.fc37.x86_64

### Output of `java -version`

openjdk 17.0.5 2022-10-18

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by llowinge**
Fix #30277

### CODE REVIEW

**Comment by geoand** [comment]
@Sgitario does most of the work on this, so I'll leave this up to him

**Review by iocanel — APPROVED** [review]
lgtm

File: extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftConfig.java
Comment (Sgitario) [inline extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftConfig.java:122]: Why not using directly the enum type? I mean, use Logger.Level instead of String.

File: extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java
Comment (Sgitario) [inline extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:528]: I think this method is only used when there is an IOException and before, we're printing the warning log: 
```java
LOG.warn("Log stream closed, redisplaying last " + LOG_TAIL_SIZE + " entries:");
```

So, I would not add this method and instead will use the Level.WARNING instead:
```
display(client.builds().withName(buildName).tailingLines(LOG_TAIL_SIZE).getLogReader(), Level.WARN);
```

So, it's consistent with the previous trace.

Comment (llowinge) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:528]: I didn't want to introduce "breaking" change. Are you OK with it ? As now it was using `INFO` level in case of IOException (so user can see potential problems).

Comment (Sgitario) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:528]: If the INFO traces are printed, then the warning traces should be too, so I don't see a regression here. 
Moreover, I think a warning trace makes more sense here than an INFO trace.

File: extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftConfig.java
Comment (llowinge) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftConfig.java]: Done. Thank you

File: extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java
Comment (llowinge) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:528]: I'm not sure if we understand each other.

This [line](https://github.com/quarkusio/quarkus/blob/main/extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java#L484) is using `INFO` and you are suggesting i should change it to `WARN` which will be breaking change, won't be ?

Comment (Sgitario) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:528]: why is it a breaking change? Even though it changes to the warning level, it will be printed.

Comment (llowinge) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:528]: Ahh, you are right. I've somehow mixed the precedence. I will change it.

**Review by Sgitario — APPROVED** [review]
Thanks!

**Comment by llowinge** [comment]
@Sgitario Not sure what `1 workflow awaiting approval` means.

**Comment by llowinge** [comment]
@Sgitario I've fixed the formatting problem.

**Pull request merged**
